internal/clients/http/image_builder/image_client.go

```diff
@@ -50,7 +50,11 @@ func (c *ibClient) Ready(ctx context.Context) error {
 		logger.Error().Err(err).Msg("Readiness request failed for image builder")
 		return err
 	}
-	defer resp.Body.Close()
+	defer func() {
+		if tempErr := resp.Body.Close(); tempErr != nil {
+			err = tempErr
```

Comment:
Fyi, we have to do this, because the defer is called after the end of the function. Therefore, if we did not use temporary error, and Close() would not return any error, the error variables would get overwritten with the 'nil' value.